src/components/application_manager/src/rpc_handler_impl.cc

```diff
@@ -218,9 +218,30 @@ bool RPCHandlerImpl::ConvertMessageToSO(
 
       rpc::ValidationReport report("RPC");
 
+      // Attach RPC version to SmartObject if it does not exist yet.
+      auto app_ptr = app_manager_.application(message.connection_key());
+      utils::SemanticVersion msg_version(2, 0, 0);
+      if (app_ptr &&
+          (output[NsSmartDeviceLink::NsJSONHandler::strings::S_PARAMS]
+               .keyExists(NsSmartDeviceLink::NsJSONHandler::strings::
+                              S_RPC_MSG_VERSION) == false)) {
+        msg_version = app_ptr->msg_version();
+        output[NsSmartDeviceLink::NsJSONHandler::strings::S_PARAMS]
+              [NsSmartDeviceLink::NsJSONHandler::strings::S_RPC_MSG_VERSION] =
+                  msg_version.toString();
+      } else if (mobile_apis::FunctionID::RegisterAppInterfaceID ==
+                 static_cast<mobile_apis::FunctionID::eType>(
+                     output[strings::params][strings::function_id].asInt())) {
+        // Assume default version 2.0.0 until properly set in RAI
+        output[NsSmartDeviceLink::NsJSONHandler::strings::S_PARAMS]
+              [NsSmartDeviceLink::NsJSONHandler::strings::S_RPC_MSG_VERSION] =
+                  msg_version.toString();
```

Comment:
Since we are setting the version manually, why not just pull it from the original `RegisterAppInterface` request instead of using `2.0.0`. Using `2.0.0` right now causes any AppHMITypes with a since of `> 2.0.0` to be rejected.

Reply:
```
auto app_version =
            output[strings::msg_params][strings::sync_msg_version];
        msg_version =
            utils::SemanticVersion(app_version[strings::major_version].asInt(),
                                   app_version[strings::minor_version].asInt(),
                                   app_version[strings::patch_version].asInt());
        output[NsSmartDeviceLink::NsJSONHandler::strings::S_PARAMS]
              [NsSmartDeviceLink::NsJSONHandler::strings::S_RPC_MSG_VERSION] =
                  msg_version.toString();
```